test/features/mailbox/repository/mailbox_respository_test.dart

```diff
@@ -229,4 +236,47 @@ void main() {
       );
     });
   });
+
+  group('[handleMailboxRightRequest] method test', () {
+    const postingRight = 'p';
+
+    setUp(() {
+      mailboxDataSource = MockMailboxDataSource();
+      mailboxCacheDataSourceImpl = MockMailboxCacheDataSourceImpl();
+      stateDataSource = MockStateDataSource();
+      mailboxRepository = MailboxRepositoryImpl(
+          {
+            DataSourceType.network: mailboxDataSource,
+            DataSourceType.local: mailboxCacheDataSourceImpl
+          },
+          stateDataSource);
+    });
+
+    test(
+      'handleMailboxRightRequest SHOULD add postingRight when action is allow and currentRights is empty', () async {
+        Map<String, List<String>?> currentRights = {};
+        MailboxRightRequest request = MailboxRightRequest(mailboxIdFixture, currentRights, MailboxSubaddressingAction.allow);
+
+        when(stateDataSource.getState(accountIdFixture, userNameFixture, StateType.mailbox))
+            .thenAnswer((_) async => StateFixtures.currentMailboxState);
+
+        when(mailboxDataSource.handleMailboxRightRequest(sessionFixture, accountIdFixture, request))
+          .thenAnswer((_) async => true);
+
+        when(mailboxDataSource.getChanges(sessionFixture, accountIdFixture, StateFixtures.currentMailboxState))
+          .thenAnswer((_) async => MailboxChangeResponse(
+              // dont really know what to put in there
+              updated: [mailboxFixture],
+              updatedProperties: Properties({'rights'})
+        ));
+
+        final response = await mailboxDataSource.getChanges(
+          sessionFixture,
+          accountIdFixture,
+          StateFixtures.currentMailboxState,
+        );
```

Comment:
But this test is meaningless. You mock it, then test the mock directly :))

Reply:
yes, it is what i said here: https://github.com/linagora/tmail-flutter/pull/3241#discussion_r1856180497

but i am about to drop [the commit with these changes](https://github.com/linagora/tmail-flutter/pull/3241/commits/767da25dd7705c096242f6718d2997024eecada0). cf the comments linked above

Reply:
done in `test/features/mailbox/data/update_rights_for_subaddressing_test.dart` !